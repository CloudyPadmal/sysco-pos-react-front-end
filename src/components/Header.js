import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

export function ConditionalHeader(P) {
    if (P.isLoggedIn) {
        return (
            <div className="collapse navbar-collapse">
                <Link className="navbar-brand"
                    style={{ paddingTop: '0px' }}
                    to="/home">Sysco PoS System</Link>
                <div className="navbar-nav mr-auto mt-2 mt-lg-0">
                    <Link className="nav-item nav-link"
                        to="/my_orders">
                        <i className="fas fa-list-ul"></i> My Orders</Link>
                    <Link className="nav-item nav-link"
                        to="/create_order">
                        <i className="fas fa-plus"></i> Create Order</Link>
                </div>
                <div className="navbar-nav my-2 my-lg-0">
                    <Link className="nav-item nav-link"
                        to="/logout">
                        <i className="fas fa-sign-out-alt"></i> Logout</Link>
                </div>
            </div>
        );
    } else {
        return (
            <div className="collapse navbar-collapse">
                <Link className="navbar-brand"
                    style={{ paddingTop: '0px' }}
                    to="/">Sysco PoS System</Link>
                <div className="navbar-nav ml-auto">
                    <Link className="nav-item nav-link"
                        to="/login">
                        <i className="fas fa-sign-in-alt"></i> Login</Link>
                    <Link className="nav-item nav-link"
                        to="/register">Register</Link>
                </div>
            </div>
        );
    }
}

/******************************************************************************
 * Navigation bar at the top of the page. This will change state depending on
 * the log state of user. The final render will be conditioned on that and will
 * be rendered using `ConditionalHeader` component in this same component class
 *****************************************************************************/
export class Header extends Component {
    render() {
        return (
            <header>
                <nav className="navbar navbar-expand-lg navbar-light bg-light">
                    <ConditionalHeader isLoggedIn={this.props.isLoggedIn} />
                </nav>
            </header>
        );
    }
}

Header.propTypes = {
    isLoggedIn: PropTypes.bool.isRequired
};

const mapStateToProps = state => ({
    isLoggedIn: state.uac.isLoggedIn
});

export default connect(mapStateToProps, null)(Header)