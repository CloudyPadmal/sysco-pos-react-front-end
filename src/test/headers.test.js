import React from 'react';
import { shallow } from './enzyme';
import { Link } from 'react-router-dom';

import Header from '../components/Header';

describe('Navigation header', () => {

  it('renders header', () => {
    const ISLOGGEDIN = true;
    const wrapper = shallow(<Header ISLOGGEDIN={ISLOGGEDIN} />);
    // Expect the wrapper object to be defined
    expect(wrapper.contains(<Link className="nav-item nav-link" to="/logout">Logout</Link>)).toBeTruthy();
    console.log(wrapper.debug());
    /* expect(wrapper.find('.item')).toHaveLength(items.length); */
  });

  /* it('renders a list item', () => {
    const items = ['Thor', 'Loki'];
    const wrapper = shallow(<List items={items} />);

    // Check if an element in the Component exists
    expect(wrapper.contains(<li key='Thor' className="item">Thor</li >)).toBeTruthy();
  });

  it('renders correct text in item', () => {
    const items = ['John', 'James', 'Luke'];
    const wrapper = shallow(<List items={items} />);

    //Expect the child of the first item to be an array
    expect(wrapper.find('.item').get(0).props.children).toEqual('John');
  }); */
});