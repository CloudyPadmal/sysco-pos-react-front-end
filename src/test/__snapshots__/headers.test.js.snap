// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav
        className="navbar navbar-expand-lg navbar-light bg-light"
      >
        <ConditionalHeader />
      </nav>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ConditionalHeader />,
        "className": "navbar navbar-expand-lg navbar-light bg-light",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "ISLOGGEDIN": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "nav",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav
          className="navbar navbar-expand-lg navbar-light bg-light"
        >
          <ConditionalHeader />
        </nav>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ConditionalHeader />,
          "className": "navbar navbar-expand-lg navbar-light bg-light",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "ISLOGGEDIN": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "nav",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
